core/keyboard_navigation_controller.ts

```diff
@@ -0,0 +1,50 @@
+/**
+ * @license
+ * Copyright 2025 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * The KeyboardNavigationController handles coordinating Blockly-wide
+ * keyboard navigation behavior, such as enabling/disabling full
+ * cursor visualization.
+ */
+export class KeyboardNavigationController {
```

Comment:
Trivial tests might be worthwhile to add to this just to provide the core regression support for toggling state & how that impacts the CSS classes.

Reply:
Done.